server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/SetMimeHeaderHandler.java

```diff
@@ -66,21 +68,23 @@ public void setHeaderValue(String headerValue) {
      * Adds header to the message
      */
     @Override
-    public HookResult onMessage(SMTPSession session, Mail mail) {
-        try {
-            MimeMessage message = mail.getMessage();
+    public Mono<HookResult> onMessage(SMTPSession session, Mail mail) {
+        return Mono.fromCallable(() -> {
+            try {
+                MimeMessage message = mail.getMessage();
 
-            // Set the header name and value (supplied at init time).
-            if (headerName != null) {
-                message.setHeader(headerName, headerValue);
-                message.saveChanges();
-            }
+                // Set the header name and value (supplied at init time).
+                if (headerName != null) {
+                    message.setHeader(headerName, headerValue);
+                    message.saveChanges();
+                }
 
-        } catch (javax.mail.MessagingException me) {
-            LOGGER.error(me.getMessage());
-        }
+            } catch (javax.mail.MessagingException me) {
+                LOGGER.error(me.getMessage());
```

Comment:
I know it is not your code but could we snik a little commit  to not loose that stacktrace? <3